config/routes.rb

```diff
@@ -0,0 +1,60 @@
+Rails.application.routes.draw do
+  get '/v1/carriers/' => 'carriers#calculate'
+  get '/v1/carriers/selected' => 'carriers#selected'
+  get '/v1/tracking' => 'carriers#tracking'
```

Comment:
I don't see a tracking method within your carriers controller. Is it supposed to be log?